### PR TITLE
MM-12355: Add CLI command "command create"

### PR DESCRIPTION
**Pull request by mojicaj**
#### Summary
Implements `command create` to create slash commands in the CLI.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9489

#### Checklist
- [x] Added or updated unit tests (required for all new features)

### CODE REVIEW

**Comment by mojicaj** [comment]
@jespino Here is the fresh pr.

**Comment by jespino** [comment]
Awesome!, very good job. I love the tests that you made :) One test is failing but except for that is ready to merge (in my opinion). I going to ask other core committer to review it (we need 2 reviews to get this merged in master).

**Comment by mojicaj** [comment]
Ok build failed and I have no idea why
```
?   	github.com/mattermost/mattermost-server/wsapi	[no test files]
Makefile:351: recipe for target 'test-te' failed
make: *** [test-te] Error 1
[TemplateProject] FAILED performing builders from: '1. Mattermost Server » mattermost-platform-template-split'
Build step 'Use builders from another project' marked build as failure
```

```
Setting status of 0c1760372946e8e3a3dcc8dff14e9da54dce0e54 to FAILURE with url https://build.mattermost.com/job/mp/job/mattermost-platform-pr-split/4183/ and message: 'Failure in Build
```

**Comment by mojicaj** [comment]
Thank you, and glad I could contribute.

**Comment by jespino** [comment]
Yes, sometimes run the tests locally is a pain. you don't need to do that, you can check our jenkins clicking on the "Details" of the build failure. There you have to go to "Console output", then the see the entire log. and search for "FAIL" with case sensitive matching, and you will find the real failing tests.

In this case:
```
--- FAIL: TestCreateCommand (2.99s)
    --- FAIL: TestCreateCommand/nil_error (0.20s)
        command_test.go:119: 
            	Error Trace:	command_test.go:119
            	Error:      	Received unexpected error:
            	            	exit status 2
            	Test:       	TestCreateCommand/nil_error
    --- PASS: TestCreateCommand/Team_not_specified (0.02s)
    --- PASS: TestCreateCommand/Team_not_found (0.20s)
    --- PASS: TestCreateCommand/Creator_not_specified (0.02s)
    --- PASS: TestCreateCommand/Creator_not_found (0.20s)
    --- PASS: TestCreateCommand/Command_not_specified (0.02s)
    --- PASS: TestCreateCommand/Trigger_not_specified (0.02s)
    --- PASS: TestCreateCommand/Blank_trigger (0.20s)
    --- PASS: TestCreateCommand/Trigger_with_space (0.20s)
    --- PASS: TestCreateCommand/Trigger_starting_with_/ (0.19s)
    --- PASS: TestCreateCommand/URL_not_specified (0.02s)
    --- PASS: TestCreateCommand/Blank_URL (0.20s)
    --- PASS: TestCreateCommand/Invalid_URL (0.20s)
    --- PASS: TestCreateCommand/Duplicate_Command (0.20s)
    --- PASS: TestCreateCommand/Misspelled_flag (0.02s)
```

File: cmd/mattermost/commands/command.go [hunk 1]
```diff
@@ -34,13 +45,92 @@ var CommandListCmd = &cobra.Command{
...
+	CommandCreateCmd.Flags().String("url", "", "Command Callback URL (required)")
+	CommandCreateCmd.MarkFlagRequired("url")
+	CommandCreateCmd.Flags().String("creator", "", "Command Creator's Username (required)")
+	CommandCreateCmd.MarkFlagRequired("creator")
+	CommandCreateCmd.Flags().String("response-username", "", "Command Response Username")
+	CommandCreateCmd.Flags().String("icon", "", "Command Icon URL")
+	CommandCreateCmd.Flags().Bool("autocomplete", false, "Show Command in autocomplete list")
+	CommandCreateCmd.Flags().String("autocompleteDesc", "", "Short Command Description for autocomplete list")
+	CommandCreateCmd.Flags().String("autocompleteHint", "", "Command Arguments displayed as help in autocomplete list")
+	CommandCreateCmd.Flags().Bool("post", false, "Command Callback URL Method Type ")
```
Comment (grundleborg) [inline cmd/mattermost/commands/command.go]: Could you change the help text for this flag to something like "Use POST method for Callback URL" as I think that makes more sense given it's a boolean flag.

**Comment by mojicaj** [comment]
Ok, made the description change.
It looks like when `make test` is run, it is trying to create cprofile log but can't for some reason.
```
testing: open /Users/jay/dev/go/src/github.com/mattermost/mattermost-server/cmd/mattermost/commands/cprofile-TestCreateCommand/nil_error-1.out: no such file or directory
```
I changed the test around to check the output text since the test passes as it should.

Can you advise if I should attempt to resolve that issue and how.

**Comment by jespino** [comment]
@mojicaj the `make test` is mainly used by the ci, the mattermost-server tests take a long to get executed properly, that the reason we don't run all of them normally in our development environment. Besides the `make test` command include some extra things like profiling which makes them slower.

My suggestion is use directly `go test -v ./cmd/mattermost/commands` and in your case, to filter your tests you can run `go test -v -run TestCreateCommand ./cmd/mattermost/commands`.

**Comment by DHaussermann** [comment]
Hi @mojicaj, 
Our team missed a detail when creating the original help wanted ticket for this. The command should not work for users who are not admins when `EnableOnlyAdminIntegrations` is set to True. Would you be able to help make this change?

**Comment by mojicaj** [comment]
@DHaussermann I'll be happy to make that change.

**Comment by mojicaj** [comment]
@DHaussermann the change has been [completed](https://github.com/mattermost/mattermost-server/pull/9734)

**Pull request merged**